gnomad_qc/v4/variant_qc/random_forest.py

```diff
@@ -0,0 +1,353 @@
+        action="store_true",
+    )
+    parser.add_argument(
+        "--test-dataset",
+        help="If test of Variant QC table should be used. (default: False)",
+        action="store_true",
+    )
+    parser.add_argument(
+        "--model_id",
+        help=(
+            "Model ID. Created by --train_rf and only needed for --apply_rf without"
+            " running --train_rf."
+        ),
+        required=False,
+    )
+
+    actions = parser.add_argument_group("Actions")
+    actions.add_argument(
+        "--list_rf_runs",
+        help=(
+            "Lists all previous RF runs, along with their model ID, parameters and"
+            " testing results."
+        ),
+        action="store_true",
+    )
+    actions.add_argument(
+        "--annotate_for_rf",
+        help="Creates an annotated HT with features for RF.",
+        action="store_true",
+    )
+    actions.add_argument("--train_rf", help="Trains RF model.", action="store_true")
+    actions.add_argument(
+        "--apply_rf", help="Applies RF model to the data.", action="store_true"
+    )
+
+    annotate_params = parser.add_argument_group("Annotate features parameters")
+    annotate_params.add_argument(
+        "--impute_features",
+        help="If set, feature imputation is performed.",
+        action="store_true",
+    )
+    annotate_params.add_argument(
+        "--n_partitions",
+        help="Desired number of partitions for annotated RF Table.",
+        type=int,
+        default=5000,
+    )
+
+    rf_params = parser.add_argument_group("Random Forest parameters")
+    rf_params.add_argument(
+        "--fp_to_tp",
+        help=(
+            "Ratio of FPs to TPs for training the RF model. If 0, all training examples"
+            " are used. (default=1.0)"
+        ),
+        default=1.0,
+        type=float,
+    )
+    rf_params.add_argument(
+        "--test_intervals",
+        help=(
+            "The specified interval(s) will be held out for testing and evaluation"
+            ' only. (default to "chr20")'
+        ),
+        nargs="+",
+        type=str,
+        default="chr20",
+    )
+    rf_params.add_argument(
+        "--num_trees",
+        help="Number of trees in the RF model. (default=500)",
+        default=500,
+        type=int,
+    )
+    rf_params.add_argument(
+        "--max_depth",
+        help="Maxmimum tree depth in the RF model. (default=5)",
+        default=5,
+        type=int,
+    )
+
+    training_params = parser.add_argument_group("Training data parameters")
+    training_params.add_argument(
+        "--adj", help="Use adj genotypes.", action="store_true"
+    )
+    training_params.add_argument(
+        "--vqsr_model_id",
+        help=(
+            "If a VQSR model ID is provided the VQSR training annotations will be used"
+            " for training."
+        ),
+        default="vqsr_alleleSpecificTrans",
+        choices=["vqsr_classic", "vqsr_alleleSpecific", "vqsr_alleleSpecificTrans"],
```

Comment:
These will need to change too since I don't think we will run the "vqsr_classic" (this is what I (or rather Laurent) called using only the aggregated site info) and maybe not even the "vqsr_alleleSpecific" (this is what we called using the allele specific, but not transmitted). I think for VQSR this time around I want to try the following:

- Allele specific using the ["quasi"](https://github.com/broadinstitute/gnomad_methods/blob/e6f042a74c91e462b77fca24d070c815e02f6f5b/gnomad/utils/sparse_mt.py#L561) method (used for UKB and v3.1) with transmitted singletons for training
- Allele specific using the aggregation of the actual [allele-specific annotations in gvcf_info](https://github.com/broadinstitute/gnomad_methods/blob/e6f042a74c91e462b77fca24d070c815e02f6f5b/gnomad/utils/sparse_mt.py#L559) with transmitted singletons for training
- Both of the above with sibling singletons added for training (in addition to the trio transmitted singletons)